eip_operator/src/controller/eip.rs

```diff
@@ -80,13 +93,207 @@ impl k8s_controller::Context for Context {
+                    .filter(|pod| pod.metadata.deletion_timestamp.is_none())
+                    .collect::<Vec<Pod>>();
+                matched_pods.sort_unstable_by_key(|s| s.name_unchecked());
+            }
+            EipSelector::Node { ref selector } => {
+                let label_selector = selector
+                    .iter()
+                    .map(|(k, v)| format!("{}={}", k, v))
+                    .collect::<Vec<String>>()
+                    .join(",");
+                matched_nodes = node_api
+                    .list(
+                        &ListParams{ label_selector: Some(label_selector), ..Default::default()}
+                    )
+                    .await?
+                    .into_iter()
+                    .filter(|node| node.metadata().deletion_timestamp.is_none())
+                    .collect::<Vec<Node>>();
+                matched_nodes.sort_unstable_by_key(|s| s.name_unchecked());
+            }
+        }
+
+        // Check to make sure our resource still matches
+        // incase our selectors have updated, or the nodes/pods we have
+        // claimed have changed
+        // let correctly_associated =
+        let mut disassociate = false;
+        match &eip.spec.selector {
+            EipSelector::Pod { pod_name: _ } => {
+                if !matched_pods.is_empty() && eip.associated_with_pod(&matched_pods[0]) {
+                    status.claim = Some(matched_pods[0].name_unchecked());
+                } else {
+                    disassociate = true;
+                    status.claim = None;
+                }
+            }
+            EipSelector::Node { selector: _ } => {
+                // check if our claim is still in the node match list
+                // if so we don't need to disassociate
+                // it's likely we'll find that we're claimed correctly, but
+                // we'll default to assuming this isn't true
+                status.claim = None;
+                disassociate = true;
+                for node in &matched_nodes {
+                    if eip.associated_with_node(node) {
+                        status.claim = Some(node.name_unchecked());
+                        disassociate = false;
+                        break;
+                    }
+                }
+            }
+        };
+
+        // Disassociaate if conditions are met
+        if disassociate {
+            info!("Disassociating EIP! {}", name);
+            let association_id = eip.association_id(&self.ec2_client).await?;
+            if let Some(id) = association_id {
+                crate::aws::disassociate_eip(&self.ec2_client, &id).await?;
+            } else {
+                info!("EIP {} was already disassociated", name);
+            }
+            status.eni = None;
+            status.private_ip_address = None;
+        }
+
+        // Find new claim, it's possible it'll match the existing
+        // association, in which case, we won't disassociate
```

Comment:
is this comment still true? it looks like the disassociation logic is already handled separately above